nemo/collections/asr/models/confidence_ensemble.py

```diff
@@ -12,22 +12,90 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from dataclasses import dataclass
 from typing import Dict, List, Optional, Union
 
 import joblib
 import numpy as np
 import torch
-from omegaconf import DictConfig, OmegaConf, open_dict
+from omegaconf import DictConfig, open_dict
 from pytorch_lightning import Trainer
 
 from nemo.collections.asr.models.asr_model import ASRModel
 from nemo.collections.asr.models.hybrid_rnnt_ctc_models import EncDecHybridRNNTCTCModel
-from nemo.collections.asr.parts.utils.asr_confidence_utils import ConfidenceConfig, get_confidence_aggregation_bank
+from nemo.collections.asr.parts.utils.asr_confidence_utils import (
+    ConfidenceConfig,
+    ConfidenceMethodConfig,
+    get_confidence_aggregation_bank,
+    get_confidence_measure_bank,
+)
 from nemo.collections.asr.parts.utils.audio_utils import ChannelSelectorType
 from nemo.core.classes import ModelPT
 from nemo.utils import model_utils
 
-__all__ = ['ConfidenceEnsembleModel']
+
+@dataclass(frozen=True)
+class ConfidenceSpec:
+    exclude_blank: bool
+    aggregation: str
+    confidence_type: str
+    alpha: float
+
+    def to_confidence_config(self) -> ConfidenceConfig:
+        if self.confidence_type == 'max_prob':
+            name = 'max_prob'
+            entropy_type = 'tsallis'  # can be any
+            entropy_norm = 'lin'  # can be any
+        else:
+            name, entropy_type, entropy_norm = self.confidence_type.split("_")
+        return ConfidenceConfig(
+            exclude_blank=self.exclude_blank,
+            aggregation=self.aggregation,
+            method_cfg=ConfidenceMethodConfig(
+                name=name, entropy_type=entropy_type, temperature=self.alpha, entropy_norm=entropy_norm,
+            ),
+        )
+
+
+def get_filtered_logprobs(transcription, exclude_blank):
```

Comment:
Possible to merge with below function, such that it is filter=True flag in below function ?

Reply:
It's being used by another function in build_ensemble.py, so that's why I separated them